File: 백준/Silver/2607. 비슷한 단어/비슷한 단어.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")


const n = Number(input[0])
const original = input[1]
let answer = 0

const target = input.slice(2)

for(let i=0;i<n-1;i++){
  let originalCnt = original
  for(let char of original){
    if(target[i].includes(char)){
      target[i] = target[i].replace(char,'')
      originalCnt = originalCnt.replace(char,'')
    }
  }
  if(originalCnt.length<=1 && target[i].length<=1){
    answer +=1
  }
}
console.log(answer)

File: 백준/Silver/2607. 비슷한 단어/비슷한 단어.py
import sys
input = sys.stdin.readline

n = int(input())
data = [list(input().strip()) for _ in range(1,n+1)]
original,*target = data
answer =0


for tar in target:
    ori = original[0:]
    t = tar[:]

    for char in tar:

      if char in ori:
        ori.remove(char)
        t.remove(char)
    if len(t) <=1 and len(ori) <=1:
        answer +=1

print(answer)
